### PR TITLE
feat: Add `Gift` component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- Adds `Gift` component.

## How it works?

image

## How to test it?

Run `yarn storybook` locally and check the [Gift component page](http://localhost:6006/?path=/docs/molecules-gift--gift).

image

### Starters Deploy Preview

[Gift - NextJS Storybook](https://nextjs-store-storybook-git-feat-fs-552-gift-component-faststore.vercel.app/?path=/docs/molecules-gift--default-story)

## References

[FS - 552 Create Gift component](https://vtex-dev.atlassian.net/browse/FS-552?atlOrigin=eyJpIjoiYzc4ZThjZGQyMGYzNDRhNmJiOGRjODBmYmY4N2ZjYzgiLCJwIjoiaiJ9)

### CODE REVIEW

File: packages/ui/src/molecules/Gift/Gift.test.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    it('`GiftContent` component should have `data-fs-gift-content` attribute', () => {
+      expect(giftContent).toHaveAttribute('data-fs-gift-content')
+    })
+  })
+
+  describe('Accessibility', () => {
+    it('should have no violations', async () => {
+      const { getByTestId } = render(<GiftTest />)
+
+      expect(await axe(getByTestId('store-gift'))).toHaveNoViolations()
```
Comment (filipewl) [inline packages/ui/src/molecules/Gift/Gift.test.tsx:49]: I'd just add this extra:
```suggestion
      expect(await axe(getByTestId('store-gift'))).toHaveNoViolations()
      expect(await axe(getByTestId('store-gift'))).toHaveNoIncompletes()
```

File: packages/ui/src/molecules/Gift/Gift.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React, { forwardRef } from 'react'
+import type { HTMLAttributes } from 'react'
+import type { ReactNode } from 'react'
+
+import Icon from '../../atoms/Icon'
+export interface GiftProps extends HTMLAttributes<HTMLDivElement> {
```
Comment (filipewl) [inline packages/ui/src/molecules/Gift/Gift.tsx:6]: Just an extra line break to separate the concerns:
```suggestion
import Icon from '../../atoms/Icon'

export interface GiftProps extends HTMLAttributes<HTMLDivElement> {
```

File: packages/ui/src/molecules/Gift/Gift.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  'aria-label'?: string
+}
+
+const Gift = forwardRef<HTMLDivElement, GiftProps>(function Gift(
+  { icon, testId = 'store-gift', children, ...otherProps },
+  ref
+) {
+  return (
+    <div ref={ref} data-fs-gift data-testid={testId} {...otherProps}>
+      {icon && <Icon component={icon} data-fs-gift-icon />}
```
Comment (filipewl) [inline packages/ui/src/molecules/Gift/Gift.tsx:28]: Is the `aria-label` prop to be used by the `Icon`?

File: packages/ui/src/molecules/Gift/Gift.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+   * ID to find this component in testing tools (e.g.: cypress,
+   * testing-library, and jest).
+   */
+  testId?: string
+  /**
+   * A React component that will be rendered as an icon.
+   */
+  icon?: ReactNode
+  /**
+   * Label to be required for accessibility.
```
Comment (filipewl) [inline packages/ui/src/molecules/Gift/Gift.tsx:17]: Just to be more clear:
```suggestion
   * Label to be required for accessibility for the Icon.
```

**Review by filipewl — COMMENTED** [review]
Looking good! Just left some minor suggestions before giving final approval.

CleanShot 2022-08-30 at 11 22 28

File: packages/ui/src/molecules/Gift/Gift.tsx (hunk 3 above)
Comment (hellofanny) [reply packages/ui/src/molecules/Gift/Gift.tsx:29]: yeah! But I'm not sure how helpful this gonna be :/

Comment (hellofanny) [reply packages/ui/src/molecules/Gift/Gift.tsx:29]: We decided to remove it! The icon will be the same, and this aria label seems not helpful.

**Review by filipewl — APPROVED** [review]

**Pull request merged**